src/RevOpts.cc

```diff
@@ -49,226 +26,115 @@ void RevOpts::SetArgs( const SST::Params& params ) {
   }
 }
 
-bool RevOpts::InitPrefetchDepth( const std::vector<std::string>& Depths ) {
-  std::vector<std::string> vstr;
-  for( uint32_t i = 0; i < Depths.size(); i++ ) {
-    std::string s = Depths[i];
+template<typename MAP>
+bool RevOpts::InitPropertyMap( const std::vector<std::string>& Opts, MAP& map ) {
+  for( auto& s : Opts ) {
+    std::vector<std::string> vstr;
+
     splitStr( s, ":", vstr );
     if( vstr.size() != 2 )
       return false;
 
-    uint32_t Core = std::stoul( vstr[0], nullptr, 0 );
-    if( Core > numCores )
+    auto Core = std::stoull( vstr[0], nullptr, 0 );
+    if( Core >= numCores )
       return false;
 
-    std::string::size_type sz          = 0;
-    uint32_t               Depth       = std::stoul( vstr[1], &sz, 0 );
+    // Store as cast integer if target is integer; otherwise store as string
+    auto parse = [&]( auto val ) {
+      if constexpr( std::is_integral_v<decltype( val )> ) {
+        map[Core] = decltype( val )( std::stoull( vstr[1], nullptr, 0 ) );
+      } else {
+        map[Core] = vstr[1];
+      }
+    };
 
-    prefetchDepth.find( Core )->second = Depth;
-    vstr.clear();
+    if constexpr( is_vector<MAP>::value ) {
+      parse( typename MAP::value_type{} );
+    } else {
+      parse( typename MAP::mapped_type{} );
+    }
   }
+
   return true;
 }
 
-bool RevOpts::InitStartAddrs( const std::vector<std::string>& StartAddrs ) {
-  std::vector<std::string> vstr;
-
+template<typename MAP>
+bool RevOpts::InitPropertyMapCores( const std::vector<std::string>& Opts, MAP& map ) {
   // check to see if we expand into multiple cores
-  if( StartAddrs.size() == 1 ) {
-    std::string s = StartAddrs[0];
+  if( Opts.size() == 1 ) {
+    auto&                    s = Opts[0];
+    std::vector<std::string> vstr;
+
     splitStr( s, ":", vstr );
     if( vstr.size() != 2 )
       return false;
 
     if( vstr[0] == "CORES" ) {
-      // set all cores to the target machine model
-      std::string::size_type sz   = 0;
-      uint64_t               Addr = std::stoull( vstr[1], &sz, 0 );
-      for( uint32_t i = 0; i < numCores; i++ ) {
-        startAddr.find( i )->second = Addr;
+
+      // set all cores to the value, stored as cast integer or as string
+      auto parse = [&]( auto val ) {
+        if constexpr( std::is_integral_v<decltype( val )> ) {
+          auto Val = decltype( val )( std::stoull( vstr[1], nullptr, 0 ) );
+          for( uint32_t i = 0; i < numCores; i++ )
+            map[i] = Val;
+        } else {
+          for( uint32_t i = 0; i < numCores; i++ )
+            map[i] = vstr[1];
+        }
+      };
+
+      if constexpr( is_vector<MAP>::value ) {
+        parse( typename MAP::value_type{} );
+      } else {
+        parse( typename MAP::mapped_type{} );
       }
+
       return true;
     }
   }
+  return InitPropertyMap( Opts, map );
```

Comment:
Here is where `InitPropertyMapCores()` falls back on `InitPropertyMap()` if no `CORES:*` is found.